File: frontend/src/types/Preferences.ts
export type ThemeModePreference = 'company_default' | 'system' | 'light' | 'dark';
export type DashboardLayoutPreference = 'compact' | 'comfortable' | 'default';
export type TableDensityPreference = 'compact' | 'comfortable';
export type FontSizePreference = 'small' | 'medium' | 'large';
export type LandingPagePreference = 'dashboard' | 'projects' | 'timesheets' | 'admin';
export type ProjectsPortfolioScopePreference = 'my_streams' | 'my_teams' | 'all';
export type ProjectsCcLayoutPreference = 'list' | 'card' | 'grouped';

export interface UserPreferences {
  id: string;
  user_id: string;
  theme_mode: ThemeModePreference;
  sidebar_expanded: boolean;
  sidebar_auto_collapse: boolean;
  dashboard_layout: DashboardLayoutPreference;
  table_density: TableDensityPreference;
  font_size: FontSizePreference;
  animations_enabled: boolean;
  reduced_motion: boolean;
  default_landing_page: LandingPagePreference;
  projects_portfolio_scope?: ProjectsPortfolioScopePreference;
  projects_cc_layout?: ProjectsCcLayoutPreference;
  projects_cc_default_view_id?: string | null;
  projects_cc_show_workstreams?: boolean;
  projects_cc_show_teams?: boolean;
  projects_cc_show_status?: boolean;
  email_notifications_enabled?: boolean;
  created_at?: string;
  updated_at?: string;
}

export const DEFAULT_USER_PREFERENCES: Omit<UserPreferences, 'id' | 'user_id'> = {
  theme_mode: 'company_default',
  sidebar_expanded: true,
  sidebar_auto_collapse: false,
  dashboard_layout: 'default',
  table_density: 'comfortable',
  font_size: 'medium',
  animations_enabled: true,
  reduced_motion: false,
  default_landing_page: 'dashboard',
  projects_portfolio_scope: 'my_streams',
  projects_cc_layout: 'grouped',
  projects_cc_show_workstreams: true,
  projects_cc_show_teams: true,
  projects_cc_show_status: true,
};

File: app/schemas/preferences.py
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.common import TimestampSchema


class UserPreferencesRead(TimestampSchema):
    id: UUID
    user_id: UUID
    theme_mode: str = "company_default"
    sidebar_expanded: bool = True
    sidebar_auto_collapse: bool = False
    dashboard_layout: str = "default"
    table_density: str = "comfortable"
    font_size: str = "medium"
    animations_enabled: bool = True
    reduced_motion: bool = False
    default_landing_page: str = "dashboard"
    projects_portfolio_scope: str = "my_streams"
    projects_cc_layout: str = "list"
    projects_cc_default_view_id: UUID | None = None
    projects_cc_show_workstreams: bool = True
    projects_cc_show_teams: bool = True
    projects_cc_show_status: bool = True
    email_notifications_enabled: bool = True
    email_assignment_enabled: bool = True
    email_reminder_enabled: bool = True
    email_ai_insights_enabled: bool = True
    email_daily_summary_enabled: bool = True
    email_weekly_summary_enabled: bool = True
    email_monthly_report_enabled: bool = True


class UserPreferencesUpdate(BaseModel):
    theme_mode: str | None = Field(default=None, max_length=32)
    sidebar_expanded: bool | None = None
    sidebar_auto_collapse: bool | None = None
    dashboard_layout: str | None = Field(default=None, max_length=20)
    table_density: str | None = Field(default=None, max_length=20)
    font_size: str | None = Field(default=None, max_length=20)
    animations_enabled: bool | None = None
    reduced_motion: bool | None = None
    default_landing_page: str | None = Field(default=None, max_length=32)
    projects_portfolio_scope: str | None = Field(default=None, max_length=32)
    projects_cc_layout: str | None = Field(default=None, max_length=20)
    projects_cc_default_view_id: UUID | None = None
    projects_cc_show_workstreams: bool | None = None
    projects_cc_show_teams: bool | None = None
    projects_cc_show_status: bool | None = None
    email_notifications_enabled: bool | None = None
    email_assignment_enabled: bool | None = None
    email_reminder_enabled: bool | None = None
    email_ai_insights_enabled: bool | None = None
    email_daily_summary_enabled: bool | None = None
    email_weekly_summary_enabled: bool | None = None
    email_monthly_report_enabled: bool | None = None
